### DISCUSSION TITLE
Config option to disable cache cleanup

### DISCUSSION THREAD

**Discussion by simonhaenisch** [renovatebot/renovate#29058] [Suggest an Idea]
## Motivation

We run multiple Renovate instances in parallel (on different repos) from the same CI machine and they share the same `cacheDir` (let me know if that's not a good idea for some reason but from the implementation looks like that should be ok).

Instances finish at different times, and the longer-running ones then try to clean up cache entries that don't exist anymore ("Error cleaning up expired file cache ...") while also probably having to re-cache some entries.

We don't persist the cache thus not cleaning up is not an issue for us.

## Proposal

Basically an option that stops the `cleanup` property from being added to the `cacheProxy` here:

https://github.com/renovatebot/renovate/blob/04f4edbd518f4fe7409a573357efed3fa8ff82a9/lib/util/cache/package/index.ts#L75

Maybe just a new global config option `cacheCleanup: boolean`?

**Comment by rarkins** [comment]
It's not intended to share a file cache amongst multiple Renovate instances

**Reply by simonhaenisch** [reply]
Ok thanks, how about SQLite (`RENOVATE_X_SQLITE_PACKAGE_CACHE`) or Redis?

**Reply by viceice** [reply]
redis can be shared, as the cleanup is done with redis ttl values

**Reply by simonhaenisch** [reply]
Ok thanks, can mark this as answered then 🙏
